extensions/commands/art/cmd_promote_graph.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os.path
+
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conans.model.recipe_ref import RecipeReference
+from conans.model.package_ref import PkgReference
+from conan.api.model import MultiPackagesList
+from conan.errors import ConanException
+
+from utils import api_request, assert_server_or_url_user_password
+from cmd_server import get_url_user_password
+
+
+def _get_path_from_rrev(rrev):
+    recipe_ref = RecipeReference.loads(rrev)
+    rrev_path = f"/_/{recipe_ref.revision}" if recipe_ref.revision else ""
+    return f"_/{recipe_ref.name}/{recipe_ref.version}{rrev_path}"
+
+
+def _get_path_from_ref(ref):
```

Comment:
Functions to map to Artifactory paths most like can be common to other commands too?